app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt

```diff
@@ -0,0 +1,180 @@
+        CrossSellingResult.Error -> "Something went wrong."
+        is CrossSellingResult.Success -> {
+            stringResource(
+                R.string.purchase_confirmation_new_insurance_today_app_state_title,
+                crossSellingResult.insuranceType
+            )
+        }
+    }
+    val descriptionText = when (crossSellingResult) {
+        CrossSellingResult.Error -> "Your purchase couldn't be completed.\nContact us in the chat."
+        is CrossSellingResult.Success -> {
+            when {
+                crossSellingResult.startingDate.dayOfMonth <= LocalDate.now().dayOfMonth -> {
+                    stringResource(R.string.purchase_confirmation_new_insurance_today_app_state_description)
+                }
+                else -> {
+                    val activationDate = crossSellingResult.startingDate.format(
+                        DateTimeFormatter.ISO_LOCAL_DATE
+                    )
+                    stringResource(
+                        R.string.purchase_confirmation_new_insurance_active_in_future_app_state_description,
+                        activationDate
+                    )
+                }
+            }
+        }
+    }
+    Column(
+        modifier = modifier,
+        verticalArrangement = Arrangement.spacedBy(18.dp),
+    ) {
+        Image(
+            painter = painterResource(drawableId),
+            contentDescription = null,
+            modifier = Modifier.size(32.dp),
+        )
+        Text(
+            titleText,
+            style = MaterialTheme.typography.h5
+        )
+        Text(
+            descriptionText,
+            style = MaterialTheme.typography.body2
+        )
+    }
+}
+
+@Composable
+private fun ButtonsSection(
+    crossSellingResult: CrossSellingResult,
+    modifier: Modifier,
+) {
+    Column(
+        modifier = modifier,
+        verticalArrangement = Arrangement.spacedBy(16.dp)
+    ) {
+        when (crossSellingResult) {
+            is CrossSellingResult.Error -> {
+                LargeContainedButton(
+                    onClick = {},
+                    modifier = Modifier.fillMaxWidth()
+                ) {
+                    Image(
+                        painter = painterResource(R.drawable.ic_chat_white),
+                        contentDescription = null,
+                    )
+                    Spacer(Modifier.width(8.dp))
+                    Text("Open chat")
+                }
+                LargeOutlinedButton(
+                    onClick = {},
+                    modifier = Modifier.fillMaxWidth()
+                ) {
+                    Text("Close")
+                }
+            }
+            is CrossSellingResult.Success -> {
+                LargeContainedButton(
+                    onClick = {},
+                    modifier = Modifier.fillMaxWidth()
+                ) {
+                    Text("Done")
+                }
+            }
+        }
+    }
+}
+
+@Preview(
+    showSystemUi = true,
```

Comment:
I'm generally a fan of not showing the system UI in previews as it takes up valuable screen real estate. What's your opinion on showing the system UI? 🙂

Reply:
In this particular case, not showing it looks like this
![image](https://user-images.githubusercontent.com/44558292/133230814-b4b9c6e9-eb26-4624-b4bc-7280b5b3dcb0.png)
and showing it looks like this
![image](https://user-images.githubusercontent.com/44558292/133230882-ae7dd6ae-929c-4885-8e9e-15cf4757b0eb.png)
That is because this screen is .fillMaxSize() anyway since it has a box filling the entire screen and the two different items are aligned to top and bottom respectively.

If anything might need changing here might be the CrossSellingResultScreen implementation if you have some ideas to do this better

Reply:
Oh, strange! Looks like the screen is less tall when system UI is shown? 😄 then the saving are moot I guess, this works totally fine 🙂